app/logic/events.py

```diff
@@ -350,24 +352,28 @@ def validateNewEventData(data):
 
     if data['timeEnd'] <= data['timeStart']:
         return (False, "Event end time must be after start time.")
-
+    
     # Validation if we are inserting a new event
     if 'id' not in data:
 
-        event = (Event.select()
-                      .where((Event.name == data['name']) &
-                             (Event.location == data['location']) &
-                             (Event.startDate == data['startDate']) &
-                             (Event.timeStart == data['timeStart'])))
+        sameEventList = list((Event.select().where((Event.name == data['name']) &
+                                                   (Event.location == data['location']) &
+                                                   (Event.startDate == data['startDate']) &
+                                                   (Event.timeStart == data['timeStart'])).execute()))
+        
+        copyList = sameEventList.copy()
+
+        for event in copyList:
+            if event.isCanceled or event.recurringId:   
+                sameEventList.remove(event)
 
         try:
             Term.get_by_id(data['term'])
         except DoesNotExist as e:
             return (False, f"Not a valid term: {data['term']}")
-
-        if event.exists():
+        if sameEventList:
             return (False, "This event already exists")
-
+        
     data['valid'] = True
```

Comment:
Does this line have any effect?

Reply:
From my understanding, the valid field is added so that when we run the saveEventToDb function, we verify that it's vent through the validation function because we check if this field is there.